File: fast-sm/src/main/java/com/fast/admin/sm/model/ZtreeModel.java
package com.fast.admin.sm.model;

import com.fast.admin.framework.request.Request;
import lombok.Data;
import org.hibernate.validator.constraints.NotBlank;

import javax.validation.constraints.NotNull;

/**
 * Created by lenovo on 2017/9/8.
 */
@Data
public class ZtreeModel extends Request {

    @NotBlank
    private Long sqlId;
    @NotBlank
    private String idKey;
    @NotBlank
    private String name;
    @NotBlank
    private String pidKey;
    @NotBlank
    private String scope;
    @NotNull
    private boolean enable;

    //点击id
    private Long id;
}

File: client/src/app/sm/model/ztree.model.ts
import { FieldModel } from './field.model';

export class ZtreeModel {
  constructor() {
    this.sqlId = '';
    this.idKey = '';
    this.name = '';
    this.pidKey = '';
    this.scope = '';
    this.enable = false;
    this.visible = false;
    this.width = 12;
    this.foreignKey = '';
    this.nodeValue = '';
  }
  visible: boolean;
  sqlId: string;
  idKey: string;
  name: string;
  pidKey: string;
  scope: string;
  enable: boolean;
  width: number;
  foreignKey: string;
  nodeValue: string;
}
